### ISSUE TITLE
The performance of build_job is bad

### ISSUE THREAD

**Issue by hubertli** [pycontribs/jenkinsapi#690]
##### ISSUE TYPE

 - Bug Report

##### Jenkinsapi VERSION
0.3.8

##### Jenkins VERSION
2.121.3

##### SUMMARY
Updated variables in example code: https://github.com/pycontribs/jenkinsapi/blob/master/examples/how_to/start_parameterized_build.py and set "lazy=True" in Jenkins.
It takes 1 minute 42 seconds to build a job. It's because build_job calls resolve_job_folders, which takes a long time to process job folders. My Jenkins server has less than 100 folders and  hundrends jobs.

Please provide a way to avoid processing job folders.

**Comment by lechat** [comment]
If the folder name and job name are known, then it is better to create `Job` object directly and then invoke it:

```
params = {'SOMETHING': 'ABC'}
jenkins = Jenkins('http://localhost')
my_job = Job(
    '%s/job/%s/job/%s' % (jenkins.baseurl, folder_name, job_name),
    job_name,
    jenkins
)
queue_item = my_job.invoke(build_params=params)
```

**Comment by hubertli** [comment]
Thanks Aleksey. Where does "Job" come from? I got the error:
Traceback (most recent call last):
  File "./jenkins.py", line 17, in <module>
    job = Job(
NameError: name 'Job' is not defined

**Comment by hubertli** [comment]
Thanks, Aleksey, but it does not work:
```
jenkins = Jenkins('http://localhost')

params = {'a': 'hello'}
folder_name = 'Test'
job_name = 'test-pipeline'

job = Job(
    '%s/job/%s/job/%s' % (jenkins.baseurl, folder_name, job_name),
    job_name,
    jenkins
)

# This will start the job and will return a QueueItem object which
# can be used to get build results
qi = job.invoke(build_params=params)

# Block this script until build is finished
if qi.is_queued() or qi.is_running():
    qi.block_until_complete()

build = qi.get_build()
print(build)
```

**Comment by hubertli** [comment]
Thanks Aleksey. It works with if "lazy=True", but the performance is still bad on ```qi.is_queued() or qi.is_running()```:
```
qi = job.invoke(build_params=params)
print('after invoking build') 

# Block this script until build is finished
if qi.is_queued() or qi.is_running():
    print('running')    # No outout
    qi.block_until_complete(interval=2)
    print('after bloking') # No output

print('getting build')
build = qi.get_build()
print(build)
```
It prints 'after invoking build' immediately, but it takes 1m23s before print 'getting build' although the job is finished. The job only run 1second.

**Comment by lechat** [comment]
This is happening because code tries to load all jobs and then get job object to get build. If I add direct job instantiation as in example above, then it will make code not compatible with folders.

Perhaps adding optional parameter to `Job` constructor - a list of folders - would be better idea, but then how do I find that information in `QueueItem`?

I will take a look on what could be done.

**Comment by lechat** [comment]
This is addressed in #699

**Comment by hubertli** [comment]
Thanks a lot! Aleksey. It works after patch your patch, it takes 15s to "get_build". It's acceptable. May I know when will you plan to release it? If it's not too long, I'll wait for using the new release, otherwise, I'll copy the files to the working area.
